$MirOS: ports/lang/llvm-gcc/patches/patch-gcc_config_gcc,v 1.1 2008/11/10 20:30:55 tg Exp $

	MidnightBSD/sparc64, MirBSD/sparc support missing
	MidnightBSD/*, MirBSD/sparc headers missing

--- gcc/config.gcc.orig	Thu Jan  1 00:00:00 1970
+++ gcc/config.gcc	Tue Nov 11 02:24:34 2008
@@ -543,6 +543,37 @@ case ${target} in
   # Assume that glibc is being used and so __cxa_atexit is provided.
   default_use_cxa_atexit=yes
   ;;
+*-*-midnightbsd*)
+  tmake_file="t-slibgcc-elf-ver t-freebsd"
+  gas=yes
+  gnu_ld=yes
+  case ${enable_threads} in
+    no)
+      fbsd_tm_file="${fbsd_tm_file} freebsd-nthr.h"
+      ;;
+    "" | yes | posix)
+      thread_file='posix'
+      tmake_file="${tmake_file} t-freebsd-thread"
+      ;;
+    *)
+      echo 'Unknown thread configuration for MidnightBSD'
+      exit 1
+      ;;
+  esac
+  fbsd_tm_file="${fbsd_tm_file} midnightbsd.h"
+  ;;
+*-*-mirbsd*)
+  tmake_file="t-libc-ok t-mirbsd t-libgcc-pic"
+  gas=yes
+  gnu_ld=yes
+  case ${enable_threads} in
+    yes)
+      thread_file='posix'
+      tmake_file="${tmake_file} t-openbsd-thread"
+      ;;
+  esac
+  extra_parts="crtbegin.o crtend.o crtbeginS.o crtendS.o crtbeginT.o"
+  ;;
 *-*-netbsd*)
   tmake_file="t-slibgcc-elf-ver t-libc-ok t-netbsd t-libgcc-pic"
   gas=yes
@@ -1113,6 +1144,20 @@ i[34567]86-*-freebsd*)
 	;;
 x86_64-*-freebsd*)
 	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h ${fbsd_tm_file} i386/x86-64.h i386/freebsd.h i386/freebsd64.h"
+	;;
+i?86-*-midnightbsd*)
+	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h"
+	tm_file="${tm_file} ${fbsd_tm_file} i386/midnightbsd.h"
+	;;
+x86_64-*-midnightbsd*)
+	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h"
+	tm_file="${tm_file} ${fbsd_tm_file} i386/x86-64.h i386/midnightbsd64.h"
+	;;
+i?86-*-mirbsd*)
+        tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h"
+	tm_file="${tm_file} mirbsd.h i386/mirbsd.h"
+	gas=yes
+	gnu_ld=yes
 	;;
 i[34567]86-*-netbsdelf*)
 	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h netbsd.h netbsd-elf.h i386/netbsd-elf.h"
